src/sagemaker_xgboost_container/callback.py

```diff
@@ -1,29 +1,27 @@
 import logging
 from smdebug.xgboost import Hook
+import xgboost as xgb
 
 
 logger = logging.getLogger(__name__)
 
-
-def add_debugging(callbacks, hyperparameters, train_dmatrix,
-                  val_dmatrix=None, json_config_path=None):
-    """Add a sagemaker debug hook to a list of callbacks.
-    :param callbacks: List of callback functions.
+class SmeDebugHook(xgb.callback.TrainingCallback, Hook):
+    """wrap sagemaker debug hook to TrainingCallback style
```

Comment:
Thanks for the change! Interested to see if this works.

```suggestion
    """Mix-in callback class. smedebug.xgboost.Hook uses legacy callback style and since XGB-3.0.0 mixing legacy callback instances with new TrainingCallback instances is not allowed. 
    
    See: https://github.com/dmlc/xgboost/blob/v1.3.0/python-package/xgboost/training.py#L92-L93
```